OptimizelySDK.Tests/EventTests/EventBuilderTest.cs

```diff
@@ -894,5 +995,371 @@ public void TestCreateImpressionEventWithBucketingIDAttribute()
+                                    new Dictionary<string, object>
+                                    {
+                                        { "decisions", new object[]
+                                            {
+                                                new Dictionary<string, object>
+                                                {
+                                                    {"campaign_id", "7719770039" },
+                                                    {"experiment_id", "7716830082" },
+                                                    {"variation_id", "7722370027" }
+                                                }
+                                            }
+                                        },
+                                        { "events", new object[]
+                                            {
+                                                new Dictionary<string, object>
+                                                {
+                                                    {"entity_id", "7719770039" },
+                                                    {"timestamp", timeStamp },
+                                                    {"uuid", guid },
+                                                    {"key", "campaign_activated" }
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            },
+                            {"attributes", new object[]
+                                {
+                                    new Dictionary<string, object>
+                                    {
+                                        {"entity_id", EventBuilder.USER_AGENT_ATTRIBUTE },
+                                        {"key", EventBuilder.USER_AGENT_ATTRIBUTE },
+                                        {"type", "custom" },
+                                        {"value", "chrome"}
+                                    },
+                                    new Dictionary<string, object>
+                                    {
+                                        {"entity_id", EventBuilder.BOT_FILTERING_ATTRIBUTE},
+                                        {"key", EventBuilder.BOT_FILTERING_ATTRIBUTE},
+                                        {"type", "custom" },
+                                        {"value", true }
+                                    }
+                                }
+                            },
+                            { "visitor_id", TestUserId }
+                        }
+                    }
+                },
+                {"project_id", "7720880029" },
+                {"account_id", "1592310167" },
+                {"client_name", "csharp-sdk" },
+                {"client_version", Optimizely.SDK_VERSION },
+                {"revision", 15 },
```

Comment:
Can you point me to where revision is set? It should be what it is in the datafile i.e. it should be string.

Reply:
https://github.com/optimizely/csharp-sdk/blob/master/OptimizelySDK/ProjectConfig.cs#L31
https://github.com/optimizely/csharp-sdk/blob/master/OptimizelySDK/ProjectConfig.cs#L266
We are deserializing it and it is set automatically.

Reply:
Bot revision and version need to be string and not int. Can you please fix that as well.

Thanks.